docs/design-documents/20241024-conduit-cli.md

```diff
@@ -0,0 +1,375 @@
+| connectors.path | path to standalone connectors' directory | No | "./connectors" |
+| db.badger.path | path to badger DB | No | "Conduit.db" |
+| processors.path | path to standalone processors' directory | No | "./processors" |
+| pipelines.path | path to the directory that has the yaml pipeline configuration files, or a single pipeline configuration file | No | "./pipelines" |
+| db.postgres.connection-string | postgres connection string | Yes |  |
+| db.postgres.table | postgres table in which to store data | No | "Conduit_kv_store" |
+| db.type | database type; accepts badger,postgres,inmemory | No | "badger" |
+| grpc.address | address for serving the gRPC API | No | ":8084" |
+| http.address | address for serving the HTTP API | No | ":8080" |
+| log.format | sets the format of the logging; accepts json, cli | No | "cli" |
+| log.level | sets logging level; accepts debug, info, warn, error, trace | No | "info" |
+| pipelines.exit-on-error | exit Conduit if a pipeline experiences an error while running | No |  |
+
+### `conduit pipelines init [NAME] [--pipelines.path] [--source] [--destination]`
+
+This command will initialize a pipeline based on the working environment. Optionally, a user could provide a different flag
+if they want to specify a different path.
+
+In the event of not being able to to read a `conduit.yaml` configuration file based on current directory or `--config.path`,
+we should prompt to set up a working Conduit environment via `conduit init`.
+
+It does not require having Conduit running.
+
+A source and a destination could be provided using the same connectors reference as described
+[here](https://conduit.io/docs/using/connectors/referencing).
+
+#### Arguments
+
+| Name | Description | Required | Default Value |
+|------|-------------|----------|---------------|
+| name  |  Pipeline file name and pipeline name  | No | `pipeline-#` (`pipeline-#.yaml`) |
+
+#### Flags
+
+| Name | Description | Required | Default Value |
+|------|-------------|----------|---------------|
+| destination  |  Plugin name of the destination connector  | No | `log` |
+| pipelines.path  |  Where to initialize a new pipeline | No | `.` (current directory) |
+| source  |  Plugin name of the source connector  | No | `generator` |
+
+#### `--help`
+
+```bash
+conduit pipelines init
+conduit pipelines init my-first-pipeline
+conduit pipelines init my-first-pipeline --pipelines.path ~/my-other-path
+conduit pipelines init --source file@v1.0 --destination file
+```
+
+### `conduit pipelines ls`
+
+This command will list the running pipelines.
+
+It requires having Conduit previously running.
+
+#### Flags
+
+| Name | Description | Required | Default Value |
+|------|-------------|----------|---------------|
+| grpc.address | address for serving the gRPC API | No | ":8084" |
+
+#### `--help`
+
+```bash
+$ conduit pipelines ls
+NAME            STATUS
+my-pipeline     running
+my-other        degraded      
+```
+
+### `conduit pipelines describe ID`
```

Comment:
I know this is about topology, but it would be good to either squeeze in the status or have a command for getting the status.

Reply:
I mean, it should return everything we could have available from a running pipeline.

Reply:
If we make the output of the command structured (e.g. JSON) we could for now offload this to another tool like `jq`.

Btw, that's one thing I like about the `gh` command tool for instance - some commands [accept the flag `--json`](https://cli.github.com/manual/gh_help_formatting) which will output JSON, while the default is human readable. This allows me to use it both manually as well as in a script.

Reply:
@lovromazgon yup. `--json` is a global flag, so that should be an option.